components/organisms/Repositories/repositories.tsx

```diff
@@ -55,7 +55,15 @@ const Repositories = ({ repositories }: RepositoriesProps): JSX.Element => {
         "/hub/insights/new"
       );
     } else {
-      signIn({ provider: "github" });
+      signIn({ 
+        provider: "github",
+        options: {
+          redirectTo: `${window.location.origin}/hub/insights/new`,
+          queryParams:{ 
```

Comment:
This is based on the supabase definition: 

```js
export type SignInWithOAuthCredentials = {
  /** One of the providers supported by GoTrue. */
  provider: Provider
  options?: {
    /** A URL to send the user to after they are confirmed. */
    redirectTo?: string
    /** A space-separated list of scopes granted to the OAuth application. */
    scopes?: string
    /** An object of query params */
    queryParams?: { [key: string]: string }
    /** If set to true does not immediately redirect the current browser context to visit the OAuth authorization page for the provider. */
    skipBrowserRedirect?: boolean
  }
}
```
Yet the query params did not work when the page redirected to insights/new 
How can I debug it? I tried to check the network tab from dev tools in Chrome but could not find this redirect. 

Is there another tool I need to try/test?

Reply:
It looks like if you pass the query params as part of the `redirectTo`, you are redirected back to the app after logging in with those query params kept intact.

```
${window.location.origin}/hub/insights/new?param=value
```

We can't pass `selectedRepos` as-is because its too much data for the URL, so they'll be some additional refactoring there